src/FSharp.FGL.IO/GDF.fs

```diff
@@ -0,0 +1,157 @@
+﻿namespace FSharp.FGL.IO
+
+open Aether
+open FSharp.FGL
+open System.Text
+
+
+module GDF =
+    
+    //Typedefinition, later used to changes the type of the associated value to the correct type.
+    type TypeInfo =
+        | VARCHAR   of string
+        | BOOLEAN   of bool
+        | DOUBLE    of float
+        | INT       of int
+
+    //Transforms string into string [], deleting quotes in the progress and splitting at commas
+    let private splitElementInfos (line:string) =
+        let chars = line.ToCharArray()
+
+        let rec stringDeconstruction insideQuote i vertexInfos (sb:StringBuilder) =
+            match chars.[i] with
+            // Handle quote marks
+            | '\'' when i = chars.Length-1  -> sb.ToString() :: vertexInfos
+            | '\'' when insideQuote         -> stringDeconstruction false  (i+1) vertexInfos (sb)
+            | '\''                          -> stringDeconstruction true (i+1) vertexInfos (sb)
+            // Handle commas
+            | ',' when insideQuote          -> stringDeconstruction insideQuote (i+1) vertexInfos (sb.Append ',')
+            | ','                           -> stringDeconstruction insideQuote (i+1) (sb.ToString() :: vertexInfos) (sb.Clear())
+            // Handle every other symbol
+            | c when i = chars.Length-1     -> sb.Append(c).ToString() :: vertexInfos
+            | c                             -> stringDeconstruction insideQuote (i+1) vertexInfos (sb.Append c)
+
+        stringDeconstruction false 0 [] (StringBuilder())
+        |> List.rev
+        |> List.toArray
+
+    //Returns a appropite function based on the value of the header, to change the corresponding values to their respective TypeDefinitoions type.
+    let private getTypeInfoMapper (headerValue:string) =
+        if      headerValue.Trim().Contains "VARCHAR"   then (fun x -> match x with |" "| "" ->  VARCHAR "DefaultValue"     |_ -> VARCHAR x)
+        elif    headerValue.Trim().Contains "INT"       then (fun x -> match x with |" "| "" ->  INT   1                    |_ -> INT    (int x))
+        elif    headerValue.Trim().Contains "DOUBLE"    then (fun x -> match x with |" "| "" ->  DOUBLE   0.0               |_ -> DOUBLE (float x))
+        elif    headerValue.Trim().Contains "BOOLEAN"   then (fun x -> match x with |" "| "" ->  BOOLEAN true               |"true" -> BOOLEAN      true|"false" -> BOOLEAN      false|_ -> failwith"unknown value in visible")
+        else failwith "unknown typeAnnotation in header"
+
+    //Returns header withput the type annotations or the def>
+    let private headerTrim (header:string) =
```

Comment:
If I see correctly, you use this function to get the name of the column. Couldn't you integrate this into the `getTypeInfoMapper` function above? The resulting functions could return both column name and the value with the according type